src/sfCommand.ts

```diff
@@ -138,10 +161,64 @@ export abstract class SfCommand<T> extends Command {
     };
   }
 
+  protected async catch(err: Error & { exitCode?: number }): Promise<Error & { exitCode?: number }> {
+    process.exitCode = process.exitCode ?? err.exitCode ?? 1;
+    this.log(this.formatError(err));
+    if (this.jsonEnabled()) {
+      CliUx.ux.styledJSON(this.toErrorJson(err));
+    }
+    return err;
+  }
+
+  /**
+   * Format errors and actions for human consumption. Adds 'Error:',
+   * When there are actions, we add 'Try this:' in blue
+   * followed by each action in red on its own line.
+   * If Error.code is present it is output last in parentheses
+   *
+   * @returns {string} Returns decorated messages.
+   */
+  protected formatError(error: Error & { actions?: string[]; code?: unknown }): string {
```

Comment:
Why not just add an `actions` and `code` property to the `Error` interface down below?

Reply:
@peternhale can you respond to this question please?

Reply:
I'll take a look at it.